src/collections/collection.ts

```diff
@@ -253,5 +256,12 @@ export class Collection {
   async dropIndexes(index: any, options: any) {
     throw new Error('Not Implemented');
   }
+}
 
+export class StargateMongooseError extends Error {
+  message: string;
+  constructor(message: any) {    
+    super(`Operation failed with the following error: ${message}`);
```

Comment:
Let's provide more info on the operation that failed. `Operation ${operationName} failed with the following error:`.

Also, add a `command` property to `StargateMongooseError` that contains the command that failed.